sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java

```diff
@@ -838,6 +850,16 @@ public void setTimestampPolicy(String timestampPolicy) {
       }
     }
 
+    /**
+     * Sets nullKeyFlag for present of null keys
+     *
+     * <p>By default, nullKeyFlag is {@code false} and will invoke {@link KafkaRecordCoder} when
+     * nullKeyFlag is {@code true}, it invokes {@link NullableKeyKafkaRecordCoder}
+     */
+    public Read<K, V> withNullKeyFlag() {
```

Comment:
I think this change will be more usable if we can make the default coder support null values instead of asking users to explicitly decide whether keys will be nullable or not during pipeline submission.

Reply:
I think the major pushback for making this as a default one is update safe and backward compatibility. Let's say that we make this by default from beam 2.33.0. Then customers with pipeline prior to 2.33.0 will not be able to update their streaming pipeline to 2.33.0 because of coder incompatibility.

Reply:
Makes sense. Probably consider renaming this to something like "supportsNullKeys()" and update documentation to describe usability instead of implementation details.

Reply:
Sure thing : )